src/components/Image/index.tsx

```diff
@@ -33,17 +31,14 @@ function Image(props) {
     useEffect(() => {
         // If an onLoad callback was specified then manually call it and pass
         // the natural image dimensions to match the native API
-        if (onLoad == null) {
+        if (onLoad == null || typeof source === 'number' || !source.uri) {
```

Comment:
NAB
I'm not sure if it's advantageous to use `onLoad == null` here. 

If onLoad turns out to be `undefined` it will also count it as equal which is not explicit enough in my opinion. Maybe we should just check if onLoad is not of type function? It's just more clear IMO.

Reply:
Great idea, done!